$NetBSD: patch-ae,v 1.1 2011/01/16 08:33:48 makoto Exp $

Fix build with png-1.5. 
 (Borrowed from editors/emacs/patches/patch-ac, and 
  probably goes away soon, when upstream will be fixed.)

--- src/image.c.orig	2011-01-16 17:00:54.000000000 +0900
+++ src/image.c	2011-01-16 17:04:35.000000000 +0900
@@ -5538,7 +5538,11 @@
   /* Avoid compiler warning about deprecated direct access to
      png_ptr's fields in libpng versions 1.4.x.  */
   image_error ("PNG error: %s", build_string (msg), Qnil);
+#if (PNG_LIBPNG_VER < 10500)
   longjmp (png_ptr->jmpbuf, 1);
+#else
+  png_longjmp (png_ptr, 1);
+#endif
 }
 
 
@@ -5700,7 +5704,7 @@
 
   /* Set error jump-back.  We come back here when the PNG library
      detects an error.  */
-  if (setjmp (png_ptr->jmpbuf))
+  if (setjmp (png_jmpbuf(png_ptr)))
     {
     error:
       if (png_ptr)
